handlers.py

```diff
@@ -0,0 +1,356 @@
+    action_name,
+    required_params=None,
+    optional_params=None,
+    skip_lock_check=False,
+):
+    """This wrapper performs error handling for the API calls.
+
+    Raises
+    ------
+    Multiple exceptions
+
+    see: https://docs.aiohttp.org/en/latest/web_exceptions.html
+    """
+    # Check parameters
+    if required_params is not None:
+        for param in required_params:
+            if param in request.query:
+                if not request.query[param]:
+                    error = {
+                        "error": {
+                            "description": "empty parameter",
+                            "parameter": param,
+                            "reason": "empty",
+                            "type": "parameter",
+                        }
+                    }
+                    raise web.HTTPUnprocessableEntity(text=json.dumps(error))
+            else:
+                error = {
+                    "error": {
+                        "description": "missing parameter",
+                        "parameter": param,
+                        "reason": "undefined",
+                        "type": "parameter",
+                    }
+                }
+                raise web.HTTPUnprocessableEntity(text=json.dumps(error))
+
+    library = request.app["tape_library"]
+    # Check that library is not locked
+    if not library.running and not skip_lock_check:
+        error = {
+            "error": {
+                "description": "Library is locked",
+                "reason": "locked",
+                "type": "lock",
+            }
+        }
+        raise web.HTTPForbidden(text=json.dumps(error))
+    # Check library queue
+    if library.check_queue_max_depth_reached():
+        error = {
+            "error": {
+                "description": "to many requests in progress",
+                "reason": "full",
+                "type": "taskqueue",
+            }
+        }
+        raise web.HTTPTooManyRequests(text=json.dumps(error))
+    # Check if action is available, run it, catch errors if any
+    if hasattr(library, "action_" + action_name):
+        try:
+            data = getattr(library, "action_" + action_name)(**request.query)
+        except web.HTTPException:
+            raise
+        except Exception as excpt:
+            logging.exception(action_name)
+            error = {
+                "error": {
+                    "description": str(excpt),
+                    "reason": "internal",
+                    "type": "server",
+                }
+            }
+            raise web.HTTPInternalServerError(text=json.dumps(error))
+
+    else:
+        error = {
+            "error": {
+                "description": "no such method",
+                "reason": "nosuch",
+                "type": "method",
+            }
+        }
+        raise web.HTTPNotImplemented(text=json.dumps(error))
+    return web.json_response(data)
+
+
+# Handlers that represent the system we simulate.
```

Comment:
The following handlers are a thin layer of coordination to keep web "stuff" out of the Library class. It also serves as the definition of the API docs. So this and swagger.yaml should be the main places we need to update to create a good ICD.